docs/extract.go

```diff
@@ -74,22 +76,42 @@ var (
 {{fencedCode .ExampleSource}}
 
 ## Parameters
+
+Here are the HCL fields that you can specify, along with their expected types
+and restrictions:
+
 {{ range .Fields}}
 - {{.Name}} ({{if .Required}}required {{end}}{{if ne .Base ""}}base {{.Base}} {{end}}{{.Type}})
 
 {{ if .MutuallyExclusive}}
-  Only one of {{codeCommaJoin .MutuallyExclusive "or"}} may be set.
+	Only one of {{codeCommaJoin .MutuallyExclusive "or"}} may be set.
 
 {{end}}{{ if .ValidValues}}
-  Valid values: {{codeCommaJoin .ValidValues "and"}}
+	Valid values: {{codeCommaJoin .ValidValues "and"}}
+
+{{end}}{{if ne .Doc ""}}  {{.Doc}}{{end}}
+{{end}}
+{{ if .HasExportedFields }}
+
+## Exported Fields
 
-{{end}}{{if ne .Doc ""}}  {{.Doc}}{{end}}{{end}}
+Here are the fields that are exported for use with 'lookup'.  Re-exported fields
+will have their own fields exported under the re-exported namespace.
+
+{{- range .GetExported}}
```

Comment:
... and this one will mean that there's no line break between the header and the exported docs. I can see that causing problems in certain markdown viewers (since empty lines specify paragraph breaks.) Maybe remove the `{{-` here?